nltk/tokenize/casual.py

```diff
@@ -262,16 +270,34 @@ def tokenize(self, text):
                               x.lower()), words))
         return words
 
+######################################################################
+# Normalization Functions
+######################################################################
+
+def reduce_lengthening(text):
+    '''
+    Replace character sequences of length 3 or greater with sequences of length 3.
+    '''
+    pattern = re.compile(r"(.)\1{2,}")
+    return pattern.sub(r"\1\1\1", text)
+
+def remove_handles(text):
+    '''
+    Remove twitter username handles from text.
+    '''
+    pattern = re.compile(r"(?<=^|(?<=[^a-zA-Z0-9-\.]))@([A-Za-z_]+[A-Za-z0-9_]+)")
```

Comment:
Yes, it comes down to personal preference, but I think this is a different
case to reserved words. Arguably, that dictum agrees with my position:
choose something with a canonical name, and if the system forbids it, mark
it deterministically, rather than choosing some arbitrary alternation (i.e.
strip <-> remove, where I would not be able to remember which is which).

On 6 July 2015 at 19:10, Pierpaolo Pantone <notifications@github.com> wrote:

> In nltk/tokenize/casual.py
> <https://github.com/nltk/nltk/pull/1035#discussion_r33916839>:
>
> > +######################################################################
> > +# Normalization Functions
> > +######################################################################
> > +
> > +def reduce_lengthening(text):
> > +    '''
> > +    Replace character sequences of length 3 or greater with sequences of length 3.
> > +    '''
> > +    pattern = re.compile(r"(.)\1{2,}")
> > +    return pattern.sub(r"\1\1\1", text)
> > +
> > +def remove_handles(text):
> > +    '''
> > +    Remove twitter username handles from text.
> > +    '''
> > +    pattern = re.compile(r"(?<=^|(?<=[^a-zA-Z0-9-\.]))@([A-Za-z_]+[A-Za-z0-9_]+)")
>
>
>    -
>
>    I removed nested look-behind and the capturing group, and I refactored
>    the regex using shorthands.
>    -
>
>    Regarding double-spaces, TweetTokenizer result should not affected by
>    them; nonetheless, I agree with you on the fact that it would be better to
>    handle them.
>    -
>
>    I personally prefer to differentiate between function and parameter
>    names. In PEP8
>    <https://www.python.org/dev/peps/pep-0008/#function-and-method-arguments>
>    something similar is suggested for reserved keywords:
>
>    If a function argument's name clashes with a reserved keyword, it is
>    generally better to append a single trailing underscore rather than use an
>    abbreviation or spelling corruption. Thus class_ is better than clss .
>    (Perhaps better is to avoid such clashes by using a synonym.)
>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nltk/nltk/pull/1035/files#r33916839>.
>